README.md

```diff
@@ -65,52 +65,429 @@ The following are the commands that you would beed to be aware and have a good h
+
+Some of the most used flags are:
+
+`df -h`: Gives output in human readable format. 
+`df -h --total`: You can use this option when all mounted filesystem are on the same system.
+
 ### `rsync`
 
+This command is used to transfer a file/directories to a remote location over a remote shell. It is fast and provides incremental transfer by only transferring the difference between source and destination. 
+
+This is pre-installed tool depending on your Linux distro, but if not it can be easily installed using your distribution package manager `sudo apt install rsync` or `sudo yum install rsync`
+
+The basic syntax of rsync command is `rsync options source destination`
+
+Some of the commonly used flags in rsysnc are: 
+
+-v: verbose
+-r: copies file recursivelt
+-a: archive mode
+-z: compress file data
+-h: human readable format
+--max-size: Set the max size for file transfer
+--include: you could use it for file extenstion '*.txt'
+--exlude: can be used to exclude file
+--remove-source-files: remove source file after rsync is complete
+--dry-run
+
+In the following example, we are copying file from existing directory into a new directory
+
+`rsync -zvh file1 fold1/`
+If the destination does not exists, rysync will create that destination folder as well. 
+
+To copy a file on a new machine, you can use the following command: 
+`rsync -avzh /fold1 root@192.168.0.1:/root/`
+This will copy the content of the folder in the home directory of root user. 
+
+To copy a file from a remote server using sshm you can use the following command: 
+`rsync -avzhe ssh root@192.168.0.1:/root/file1 /tmp`
+
+It's highly recommended to use dry run before you rsync, the output will tell you what will be synced across:
+`rsync --dry-run -zvh source destination`
 ### `top`
 
+This command will show you the list of Linux processes. This is supposed to be a summary page and as soon as you type `top`, it will show you a dashboard of running processes in the Linux Kernel. You can type q to quit from the summary page. 
+
+The following columns are shown in the top dashboard: 
+
+-PID: Task unique process ID
+-PR: Process priority, the lower the number the higher is the priority
+-VIRT: Total virtual memory used by the task
+-USER: Owner of the task
+-%CPU: CPU usage
+-TIME: CPU time
+-SHR: Shared memory in KB used by a task
+-NI: Nice value of a task, a negative NIC implies higher priority and positive NI represents lower priority. 
+-%MEM: Memory usage of a task
+-RES: How much physical RAM is the process using
+-COMMAND: The name of the command that stated the process. 
+
+This [link](https://www.redhat.com/sysadmin/interpret-top-output) has all the detailed informaton on what the first 4 lines of the summary page. 
+
+Some of the most used commands are listed below
+
+`top -u suhail`
+This command will list all the process for the user Suhail
+
+`top -s`
+Runs in secure mode
```

Comment:
May be explain, what does it mean when top is running on secure mode. 

Please also add section called `Things to know` stating that when running in secure mode not all of the interactive commands are available like '*'

Reply:
This has been addressed as well